protocols/gossipsub/src/behaviour.rs

```diff
@@ -1857,29 +1863,20 @@ where
         for subscription in filtered_topics {
             // get the peers from the mapping, or insert empty lists if the topic doesn't exist
             let topic_hash = &subscription.topic_hash;
-            let peer_list = self.topic_peers.entry(topic_hash.clone()).or_default();
 
             match subscription.action {
                 SubscriptionAction::Subscribe => {
-                    if peer_list.insert(*propagation_source) {
+                    if peer.topics.insert(topic_hash.clone()) {
                         tracing::debug!(
                             peer=%propagation_source,
                             topic=%topic_hash,
                             "SUBSCRIPTION: Adding gossip peer to topic"
                         );
                     }
```

Comment:
Should we add `m.inc_topic_peers(topic_hash);` here?

```suggestion
                    if peer.topics.insert(topic_hash.clone()) {
                        tracing::debug!(
                            peer=%propagation_source,
                            topic=%topic_hash,
                            "SUBSCRIPTION: Adding gossip peer to topic"
                        );
                    }

                    if let Some(m) = self.metrics.as_mut() {
                        m.inc_topic_peers(topic_hash);
                    }
```

Reply:
yup makes sense, also updated the other place where we inc the metrics to only do it when we actually insert in `self.topics`.
Thanks Akihito!